.github/workflows/nodejs.yml

```diff
@@ -30,19 +28,32 @@ jobs:
     needs: build
     runs-on: ubuntu-latest
     steps:
+
     - name: Checkout Repository
       uses: actions/checkout@v3
+
     - name: Use Node.js
       uses: actions/setup-node@v3
       with:
         node-version: 18
+
     - name: Install npm-force-resolutions
       run: npm install -g npm-force-resolutions
+
     - name: Install dependencies
       run: npm install
+
     - name: Build Package
       run: npm run build --if-present
+
     - name: Tag Version
       run: npm --no-git-tag-version version prerelease --preid=${GITHUB_REF##*/}-${GITHUB_SHA:0:8}
+
+    - name: Configure Project-Level .npmrc File
+      run: npm config set --location=project @adzerk:registry=https://'${NPM_REGISTRY}' //'${NPM_REGISTRY}'/:_authToken='${NPM_SECRET}'
```

Comment:
It looks like these rely on NPM_REGISTRY and NPM_SECRET, don't they? Am I right in assuming that this step won't have those values, as they've been moved into the step below instead of living at the top level?

Reply:
Normally yes, but what I've done is wrap the shell variables in single quotes as the actual values will be substituted in by npm when `npm publish` is run.

Reply:
`npm publish` then substitutes those values in from env variables when run